src/realm/array_direct.hpp

```diff
@@ -233,7 +233,7 @@ class bf_ref;
 class bf_iterator {
     uint64_t* data_area;
     uint64_t* first_word_ptr;
-    size_t field_position;
+    uint64_t field_position;
```

Comment:
this is just a slowdown on 32 bit machines. The position is never too large for a 32 bit value.